src/main/kotlin/domain/Game.kt

```diff
@@ -0,0 +1,33 @@
+package domain
+
+import domain.card.CardDeck
+import domain.card.ShuffledCardDeck
+
+class Game(val players: List<Player>, private val cardDeck: CardDeck = ShuffledCardDeck.createNew()) {
+
+    fun start() {
+        players.hitInitialCards()
+    }
+
+    fun playersCanReceiveMoreCard(): List<Player> {
+        return players.filter {
+            it.canReceiveMoreCard()
+        }
+    }
+
+    fun hit(player: Player) {
+        player.hit(cardDeck.pop())
+    }
+
+    private fun List<Player>.hitInitialCards() {
+        forEach {
+            it.hitInitialCards()
+        }
+    }
+
+    private fun Player.hitInitialCards() {
+        repeat(2) {
+            hit(cardDeck.pop())
+        }
+    }
```

Comment:
확장 함수는 기존 클래스의 변경 없이도 클래스의 기능을 확장할 수 있게 만들어주는 좋은 도구입니다.
하지만 확장 함수를 남용하게 되면 객체가 원래 해야할 책임/역할이 코드 곳곳에 분산되어 객체의 응집도가 낮아질 수 있어요.
위 경우는 확장함수로 표현하지 않고도 충분히 가독성 있는 코드가 될 것 같아요.
```suggestion
    fun start() {
        players.onEach { 
            it.hit(cardDeck.pop() 
            it.hit(cardDeck.pop() 
        }
    }
    // 이하 생략
```